cmd/kubeadm/app/preflight/checks.go

```diff
@@ -650,7 +650,6 @@ func RunInitMasterChecks(cfg *kubeadmapi.MasterConfiguration) error {
 		PortOpenCheck{port: 10252},
 		HTTPProxyCheck{Proto: "https", Host: cfg.API.AdvertiseAddress, Port: int(cfg.API.BindPort)},
 		DirAvailableCheck{Path: filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.ManifestsSubDirName)},
-		DirAvailableCheck{Path: "/var/lib/kubelet"},
```

Comment:
I think it will be a lot better if DirAvailableCheck will be enhanced to do filepath.Walk() and see if there are files present. It should fail check only if directory contains files, but should tolerate if there are empty sub-dirs.

Reply:
>  It should fail check only if directory contains files, but should tolerate if there are empty sub-dirs.

not really... the kubelet is running in the background, so `kubeadm` should not expect its runtime dir to be empty.

Reply:
Preflight checks are executed at the time when kubelet has no valid configuration, so should not be really running. And even if it running, it should be nothing in `/var/lib/kubelet/pods`, which might mean leftovers from previous configurations if `kubeadm reset` was not properly executed.

As alternative, we can add check that `/var/lib/kubelet/pods` and `/var/lib/kubelet/pki` are empty.

Reply:
@luxas asked that rather than remove these, I change these to check the pod manifest subfolder specifically, since that is what kubeadm writes to. will update.

Reply:
> @luxas asked that rather than remove these, I change these to check the pod manifest subfolder specifically, since that is what kubeadm writes to. will update.

actually, I don't think that's a good idea... the kubelet owns the runtime dir. `kubeadm` can check the paths it writes to (pod manifests, etc), but shouldn't make assumptions about what the kubelet writes to its runtime dir. I think this is good as-is.

Reply:
What would happen on nodes, where there are no static manifests, but kublet was not properly reset and still running some pods and having certs from previous node setup ?

Reply:
> As alternative, we can add check that /var/lib/kubelet/pods and /var/lib/kubelet/pki are empty.

`/var/lib/kubelet/pki` will not be empty if the kubelet is crashlooping in the background... it generates serving certificates there

Reply:
I'm ok for now to remove that directory check, as files are generated by crashlooping kubelet. In 1.9 I think we need to have separate issue that would be handling kubelet start/stop in better way in init/join commands. Right now it seems that `kubeadm` has even more issues, like kubelet will not be started if user did `kubeadm reset` then `kubeadm join --skip-preflight-checks`.